### PR TITLE
kinder: remove the handling of 1.25 and older versions of k8s

### PR DESCRIPTION
**Pull request by neolit123**
- Remove the repository replace from k8s.gcr.io -> registry.k8s.io
- Remove "kubelet-config-x.yy" handling, use "kubelet-config"
- Stop untainting "master" taints, only untaint control-plane taints
- Stop handling --experimental-patches, use --patches

xref https://github.com/kubernetes/kubeadm/issues/2910

### CODE REVIEW

File: kinder/pkg/cluster/manager/actions/kubeadm-init.go [hunk 1]
```diff
@@ -97,9 +95,7 @@ func kubeadmInit(cp1 *status.Node, copyCertsMode CopyCertsMode, patchesDir, igno
 		)
 	}
 	if patchesDir != "" {
-		if cp1.MustKubeadmVersion().LessThan(constants.V1_22) {
-			initArgs = append(initArgs, "--experimental-patches", constants.PatchesDir)
-		}
+		initArgs = append(initArgs, "--patches", constants.PatchesDir)
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/kubeadm-init.go:98]: --patches was completely missing in some places

Comment (pacoxu) [reply kinder/pkg/cluster/manager/actions/kubeadm-init.go:98]: IIRC, this is in the kubeadm config `/kind/kubeadm.conf` in new versions.
```
patches:
  directory: /kinder/patches
```

Comment (neolit123) [reply kinder/pkg/cluster/manager/actions/kubeadm-init.go:98]: right..thanks for the reminder.
only upgrade needs --patches

Comment (neolit123) [reply kinder/pkg/cluster/manager/actions/kubeadm-init.go]: updated!

**Comment by pacoxu** [comment]
> will hold this until 1.29 is released.

/lgtm

File: kinder/pkg/cluster/manager/actions/kubeadm-init.go [hunk 2]
```diff
@@ -229,13 +212,7 @@ func postInit(c *status.Cluster, wait time.Duration) error {
...
-		// TODO: Once kubeadm 1.23 is no longer supported remove the <1.24 handling.
-		// TODO: Remove only the "control-plane" taint for kubeadm >= 1.25.
-		// https://github.com/kubernetes/kubeadm/issues/2200
-		taints := []string{"node-role.kubernetes.io/control-plane-", "node-role.kubernetes.io/master-"}
-		if cp1.MustKubeadmVersion().LessThan(version.MustParseSemantic("v1.24.0")) {
-			taints = []string{"node-role.kubernetes.io/master-"}
-		}
+		taints := []string{"node-role.kubernetes.io/control-plane-"}
 		taintArgs := []string{"--kubeconfig=/etc/kubernetes/admin.conf", "taint", "nodes", "--all"}
 		taintArgs = append(taintArgs, taints...)
```
Comment (chendave) [inline kinder/pkg/cluster/manager/actions/kubeadm-init.go:217]: no longer to be a slice, fine as is.
```suggestion
		taint := "node-role.kubernetes.io/control-plane-"
		taintArgs := []string{"--kubeconfig=/etc/kubernetes/admin.conf", "taint", "nodes", "--all"}
		taintArgs = append(taintArgs, taint)
```

File: kinder/pkg/cluster/manager/actions/kubeadm-join.go [hunk 3]
```diff
@@ -97,11 +97,6 @@ func kubeadmJoinControlPlane(cp *status.Node, patchesDir, ignorePreflightErrors
 		fmt.Sprintf("--ignore-preflight-errors=%s", ignorePreflightErrors),
 		fmt.Sprintf("--v=%d", vLevel),
 	}
```
Comment (chendave) [inline kinder/pkg/cluster/manager/actions/kubeadm-join.go:99]: I cannot comment on Line93, a quick question, does the parameter `patchesDir` is still needed, if it's only read from config file, I think we can remove it from func signature.

**Review by chendave — COMMENTED** [review]
/lgtm

nice cleanup!

**Pull request merged**